share/disambiguation.py

```diff
@@ -131,9 +134,12 @@ def _instance_for_node(self, node):
                 return None
             if len(found) == 1:
                 return found[0]
-            if all_query.children or any('__' in str(query) for query in queries):
+            if all_query.children:
                 logger.warning('Multiple %ss returned for %s (The main query) bailing', concrete_model, all_query)
                 break
+            if all('__' in str(query) for query in queries):
```

Comment:
Yeah, that was exactly the situation that made me change it. I had an organization with 2 identifiers and a name, and because there was a person with the same name, the query had two results. With `all` here, it'll narrow the search by type in the next loop iteration and correctly disambiguate to the matching organization. If the query didn't use a field on the model, though, there'd be no point narrowing by type, which is why `all` seemed right.